### ISSUE TITLE
v23.11.2 (and v23.11.3) - cannot login - stuck in upgrading to a new version

### ISSUE THREAD

**Issue by neyrond** [jlesage/docker-crashplan-pro#443]
### Current Behavior

After upgrading to the latest version I cannot login to UI, getting the followings message after entering credentials: `Upgrading to a new version. You can sign in once the upgrade completes`. This situation continues for several days.

`service.log.0` gives me that:
```
 backup42.service.ui.UIController] UI:: UserActionRequest: LoginRequestMessage[*******] LOGIN: username=******, password=****.....
 backup42.service.ui.UIController] UI:: UserActionResponse: LoginResponseMessage@******[ errors=[CPErrors.Global.UPGRADING : []]session=******
```

### Environment

Device: DS918+

### Container creation

docker run \
-d \
--privileged \
--name=crashplan-pro \
--restart=unless-stopped  \
-v /volume1/docker-appdata/crashplan-pro:/config:rw \
-v /volume1:/volume1 \
-v CLEAN_TMP_DIR=1 \
-p 5800:5800 \
-e CRASHPLAN_SRV_MAX_MEM=4G \
-e USER_ID=0 -e GROUP_ID=0 \
-e KEEP_APP_RUNNING=1 \
jlesage/crashplan-pro:latest

### Container log

```text
╭――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――╮
    │                                                                      │
    │ Application:           CrashPlan for Small Business                  │
    │ Application Version:   11.2.0                                        │
    │ Docker Image Version:  23.11.3                                       │
    │ Docker Image Platform: linux/amd64                                   │
    │                                                                      │
    ╰――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――╯
[cont-init   ] 89-info.sh: terminated successfully.
[cont-init   ] all container initialization scripts executed.
[init        ] giving control to process supervisor.
[supervisor  ] loading services...
[supervisor  ] loading service 'default'...
[supervisor  ] loading service 'app'...
[supervisor  ] loading service 'gui'...
[supervisor  ] loading service 'certsmonitor'...
[supervisor  ] service 'certsmonitor' is disabled.
[supervisor  ] loading service 'nginx'...
[supervisor  ] loading service 'xvnc'...
[supervisor  ] loading service 'openbox'...
[supervisor  ] loading service 'CrashPlanEngine'...
[supervisor  ] loading service 'logmonitor'...
[supervisor  ] loading service 'logrotate'...
[supervisor  ] all services loaded.
[supervisor     ] starting services...
[supervisor     ] starting service 'xvnc'...
[xvnc           ] Xvnc TigerVNC 1.13.1 - built Nov 17 2023 22:53:44
[xvnc           ] Copyright (C) 1999-2022 TigerVNC Team and many others (see README.rst)
[xvnc           ] See https://www.tigervnc.org for information on TigerVNC.
[xvnc           ] Underlying X server release 12014000
[xvnc           ] Tue Nov 21 12:07:48 2023
[xvnc           ]  vncext:      VNC extension running!
[xvnc           ]  vncext:      Listening for VNC connections on /tmp/vnc.sock (mode 0660)
[xvnc           ]  vncext:      Listening for VNC connections on all interface(s), port 5900
[xvnc           ]  vncext:      created VNC server for screen 0
[supervisor     ] starting service 'nginx'...
[nginx          ] Listening for HTTP connections on port 5800.
[supervisor     ] starting service 'openbox'...
[supervisor     ] starting service 'CrashPlanEngine'...
[supervisor     ] starting service 'app'...
[supervisor     ] starting service 'logmonitor'...
[logmonitor     ] Monitoring log file: /config/log/service.log.0
[logmonitor     ] Monitoring status file: /config/log/app.log
[supervisor     ] all services started.
[xvnc           ] Tue Nov 21 12:08:42 2023
[xvnc           ]  Connections: accepted: /tmp/vnc.sock
[xvnc           ]  SConnection: Client needs protocol version 3.8
[xvnc           ]  SConnection: Client requests security type None(1)
[xvnc           ]  VNCSConnST:  Server default pixel format depth 24 (32bpp) little-endian rgb888
[xvnc           ]  VNCSConnST:  Client pixel format depth 24 (32bpp) little-endian bgr888
```

**Comment by carmalius** [comment]
Removed cache `/crashplan-pro/cache/*`, still getting the same error: `CPErrors.Global.UPGRADING `

**Comment by neyrond** [comment]
Removed cache `/crashplan-pro/cache/*`, still getting the same error: `CPErrors.Global.UPGRADING `

**Comment by tchebyte** [comment]
> Is this still an issue ? It's not clear to me if this error is related to upgrade of the CrashPlan server...

same problem here.

`INFO  34715192-132 backup42.service.ui.UIController] UI:: UserActionResponse: LoginResponseMessage@********[ errors=[CPErrors.Global.UPGRADING : []]session=********]{}`

**Comment by jlesage** [comment]
Did you try to contact the CrashPlan support team about this ?
If you do, please report back here about the solution/problem.

**Comment by tchebyte** [comment]
> Did you try to contact the CrashPlan support team about this ? If you do, please report back here about the solution/problem.

I did. They replied:

> If your device is stuck in this state, it is typically resolved by uninstalling and reinstalling the application manually. Please follow the Standard Uninstall steps listed below:
> [Uninstalling and Reinstalling the CrashPlan App > Standard Uninstall](https://support.crashplan.com/hc/en-us/articles/9016646573069--Uninstall-and-reinstall-the-CrashPlan-app#01GCES6ES6ZBZKCXEZ7G6SK2BS)
>  
> After uninstalling, you can download the latest version of the CrashPlan app by signing into the [CrashPlan console](https://support.crashplan.com/hc/en-us/articles/9026076856333) and selecting Downloads on the top of the page, then selecting the installer appropriate for your operating system.

I'm new with docker. What should I to achieve this?

**Comment by neyrond** [comment]
> follow the Standard Uninstall steps listed below

I just went through `uninstall.sh` in `CrashPlanSmb_11.2.0_520_Linux.tgz` archive. It looks in standard uninstall they remove some binaries, some config files, but leavings most of the user data:

```
    rm -f "${BIN_DIR}/${APP_BASENAME}Desktop"
    rm -f "/usr/local/bin/${APP_BASENAME_LOWER}"
...
    rm -f "${APP_DIR}/conf/default.service.xml"
    rm -f "${APP_DIR}/conf/service.log.xml"
...

```
I haven't researched it deeply, maybe I'm missing something important. 

I tried to replace default.service.xml and service.log.xml in Crashplan Docker appdata to the ones from the last version, but it didn't help.

**Comment by jlesage** [comment]
I guess you can try to start from scratch and adopt your backup:
* Stop the container. 
* Rename the directory you mapped to `/config` (e.g. `mv /volume1/docker-appdata/crashplan-pro /volume1/docker-appdata/crashplan-pro.bk`).
* Start the container.

You will have the option to adopt an existing device/backup to avoid re-uploading everything.

**Comment by tchebyte** [comment]
Thanks, that worked!
I had to preserve conf/default.service.xml in the new /config folder to login though.

**Comment by tchebyte** [comment]
The first login page asked for "server name". I tried "central.crashplanpro.com:4287", "console.us2.crashplanpro.com:4287", "clients.us2.crashplan.com:4287", "console.us5.crashplanpro.com:4287", "clients.us5.crashplan.com:4287", but no luck.

So, I restored this file and I could login and I took over the backup as [this instructions](https://github.com/jlesage/docker-crashplan-pro#taking-over-existing-backup).
